sphinx/screens-detail/subscription/subscription/src/main/java/chat/sphinx/subscription/ui/SubscriptionFragment.kt

```diff
@@ -0,0 +1,257 @@
+                radioButtonSubscriptionAmountCustom, editTextSubscriptionCustomAmount
+            )
+
+            radioGroupSubscriptionEndRule.setOnCheckedChangeListenerWithInputInteraction(
+                radioButtonSubscriptionMakeQuantity, editTextSubscriptionMakeQuantity
+            )
+
+            radioGroupSubscriptionEndRule.setOnCheckedChangeListenerWithDatePickerInputInteraction(
+                radioButtonSubscriptionPayUntil, editTextSubscriptionPayUntil
+            )
+
+            buttonSubscriptionSave.setOnClickListener {
+
+                val amount: Sat? = when (radioGroupSubscriptionAmount.checkedRadioButtonId) {
+                    R.id.radio_button_subscription_amount_500_sats -> {
+                        Sat(500)
+                    }
+                    R.id.radio_button_subscription_amount_1000_sats -> {
+                        Sat(1000)
+                    }
+                    R.id.radio_button_subscription_amount_2000_sats -> {
+                        Sat(2000)
+                    }
+                    R.id.radio_button_subscription_amount_custom -> {
+                        editTextSubscriptionCustomAmount.text?.toString()?.toLongOrNull()?.let {
+                            Sat(it)
+                        }
+                    }
+                    else -> null
+                }
+
+                val cron: String? = when (radioGroupSubscriptionTimeInterval.checkedRadioButtonId) {
+                    R.id.radio_button_subscription_daily -> {
+                        "daily"
+                    }
+                    R.id.radio_button_subscription_weekly -> {
+                        "weekly"
+                    }
+                    R.id.radio_button_subscription_monthly -> {
+                        "monthly"
+                    }
+                    else -> null
+                }
+
+                var endNumber: Long? = null
+                val endDate: DateTime? = when (radioGroupSubscriptionEndRule.checkedRadioButtonId) {
+                    R.id.radio_button_subscription_make_quantity -> {
+                        editTextSubscriptionMakeQuantity.text?.toString()?.let {
+                            endNumber = it.toLongOrNull()
+                        }
+                        null
+                    }
+                    R.id.radio_button_subscription_pay_until -> {
+                        calendar.timeInMillis.toDateTime()
+                    }
+                    else -> null
+                }
+
+                viewModel.saveSubscription(
+                    amount,
+                    cron,
+                    endDate,
+                    endNumber
+                )
+            }
+            (requireActivity() as InsetterActivity).addNavigationBarPadding(layoutConstraintSubscription)
+        }
+
+        viewModel.initSubscription()
+    }
+
+    override suspend fun onViewStateFlowCollect(viewState: SubscriptionViewState) {
```

Comment:
The primary purpose of a `ViewState` is to describe the state of the view, not only hold the data and then based off of the data, set the view. As it looks now, one cannot look at the xml layout, and the view state, and decipher what each view's state is. All of the logic is in the renderer when it should belong in the ViewModel (or when the view's state is instantiated for immutability/encapsulation purposes).